### PR TITLE
Reflect scatter3d marker opacity reality in attributes

### PR DESCRIPTION
**Pull request by etpinard**
@cldougl @mdtusz 

`marker.opacity` in scatter3d traces must a be scalar. That is, it doesn't support arrays of scalar like its `scatter` counterpart. We made this decision for performance reasons. Marker opacity arrays would require an extra pass over the data on every render loop.

Before this PR, this coercion was made [here](https://github.com/plotly/plotly.js/blob/master/src/traces/scatter3d/convert.js#L322-L325) but wasn't reflected in the scatter3d trace attributes.

### CODE REVIEW

File: src/traces/scatter3d/attributes.js [hunk 1]
```diff
@@ -114,7 +114,17 @@ module.exports = {
...
-        opacity: scatterMarkerAttrs.opacity,
+        opacity: extendFlat({}, scatterMarkerAttrs.opacity, {
+            arrayOk: false,
+            description: [
+                'Sets the marker opacity.',
+                'Note that the marker opacity for scatter3d traces',
+                'must be a scalar value for performance reasons.',
+                'To set a blending opacity value',
+                '(i.e. which is not transparent), set *marker.color*',
+                'to an rgba color and use its alpha channel.'
```
Comment (etpinard) [inline src/traces/scatter3d/attributes.js]: @cldougl I could use a proof reader.

File: src/traces/scatter3d/convert.js [hunk 2]
```diff
@@ -319,10 +319,9 @@ proto.update = function(data) {
         this.linePlot = null;
     }
 
+    // N.B. marker.opacity must be a scalar for performance
     var scatterOpacity = data.opacity;
-    if(data.marker && typeof data.marker.opacity === 'number') {
-      scatterOpacity *= data.marker.opacity;
-    }
+    if(data.marker && data.marker.opacity) scatterOpacity *= data.marker.opacity;
```
Comment (etpinard) [inline src/traces/scatter3d/convert.js]: Now `data.marker.opacity` will always be a scalar after the `coerce` step.

File: src/traces/scatter3d/attributes.js (hunk 1 above)
Comment (cldougl) [inline src/traces/scatter3d/attributes.js]: Nice- looks great to me!

**Comment by cldougl** [comment]
:dancer:

**Pull request merged**